mono/mini/interp/transform.c

```diff
@@ -1648,7 +1648,9 @@ interp_handle_intrinsics (TransformData *td, MonoMethod *target_method, MonoClas
 			td->ip += 5;
 			return TRUE;
 		}
-	} else if (in_corlib && !strcmp (klass_name_space, "Internal.Runtime.CompilerServices") && !strcmp (klass_name, "Unsafe")) {
+	} else if (((in_corlib && !strcmp (klass_name_space, "Internal.Runtime.CompilerServices"))
+				|| !strcmp (klass_name_space, "System.Runtime.CompilerServices"))
```

Comment:
Parens to group the `||`?
Actually it doesn't look correct anyway.
It looks like `in_corlib` is not guarding as much as it should.

Reply:
@jaykrell I assume on netcore these intrinsics are found also in a different assembly for public use

Reply:
Yeah I wondered that, could be.
A shame to strcmp so often imho.